terraform/eval_validate.go

```diff
@@ -101,6 +121,64 @@ func (n *EvalValidateProvisioner) Eval(ctx EvalContext) (interface{}, error) {
 	}
 }
 
+func (n *EvalValidateProvisioner) validateConnConfig(connConfig *ResourceConfig) (warns []string, errs []error) {
```

Comment:
That's exactly what I tried, but then that adds a dependency from `terraform` to `communicator`, which is problematic. In order for us to call from `terraform` to `communicator` we need to eliminate all of the `terraform` dependencies from `communicator` itself, which I think is possible (it seems to use it only because currently communicators get passed a full `terraform.InstanceState` on init) but a pretty disruptive interface change for a relatively-minor UX improvement.